### DISCUSSION TITLE
Remove focus

### DISCUSSION THREAD

**Discussion by mblack101** [AvaloniaUI/Avalonia#5707] [Q&A]
How I can remove the keyboard focus from a TextBox (possibly not by focusing another control)?

**Comment by maxkatz6** [comment]
FocusManager.Instance.Focus(null), which will focus window root iirc or focus window root directly.
It's not possible to "delete" focus completely from the application, and it doesn't makes much sense.

Or do you probably just need to remove focus visual? If yes, you can try to use ":focus" and ":focus-visible" presuclasses and override border properties. Use Default/Fluent theme styles for TextBox as a reference.

**Reply by mblack101** [reply]
I tried using Application.Current.FocusManager.Focus(null); but that didn't seem to do anything (focus remains on my TextBox).

**Reply by maxkatz6** [reply]
It was fixed in master and will be available with next release
https://github.com/AvaloniaUI/Avalonia/pull/5492

**Reply by trampster** [reply]
I just tried this fix, but Application.Current.FocusManager.Focus doesn't exist. There is a Application.Current?.Focus(new FocusRequest());but it does nothing.

**Reply by maxkatz6** [reply]
> Application.Current?.Focus

I don't know where this is from... I might be a leaking extension method though.

> but Application.Current.FocusManager.Focus doesn't exist

See `Window/TopLevel.FocusManager`.
